### PR TITLE
Functional Registration/Aliasing, Tracking, and Listing

### PR DESCRIPTION
**Pull request by Joacchim**
This PR provides the following features, along with the necessary fixes:
 - Sidebar mode switch between "Browse" (Supposed to allow to jump on the right URL for the comic/source) and "Register" (allows creating new entries + adding sources to existing entries)
 - A new "Register" Button in the sidebar used to validate the creation of a new entry
 - Functional registration of a Comic into the web-extension's storage
 - Functional Aliasing of new sources to an existing Comic from the web-extension's storage
 - Functional Listing of the saved Comics & Sources
 - Visual Notification (Green fading to default color for success, Red fading to default color for failure) through the toggle button, to notify the user of the latest operation's status. Includes:
   - Registration
   - Aliasing
   - Auto-Tracking

Here is the registration process in screenshots:
1. Go to your Comic's page (on a supported reader)
register_init_state
2. Click on the round `+` button to start the registration process
register_sidebar
3. Use the searchbox to search for your Comic
register_sidebar_search
4. Finalize Registration
4.1. You found it in your tracking list ! Click on the name of the Manga you with to add a source to.
[No Screenshot]
4.2. You could not find it ! Create a new tracking entry by clicking on the "+" button on the side of the searchbox (Hovering over it highlights it)
register_sidebar_click

Tracking visual notification:
track_ui

Browsing the registered entries in the sidebar (BROWSE mode)
browse_sidebar

Note that these following screenshots show a few things that remain to do:
Cleanups:
 - Register round Button is still showing after Registration is completed (memoization of the ComicID by the BmcEngine object prevents an update). We need to force this update.

User Experience:
 - The two `+` buttons for registration feel like an unecessary double. We could remove the Infobar completely, and only use the new Registration button instead.
 - Searchbox isn't filled by default when attempting to Register. It could be a cool addition
 - Searchbox content isn't emptied once the SideBar is hidden
 - Searchbox content (if not emptied) is not taken into account for filtering of the Comics when reopening the Sidebar

I've also identified, while testing this, some odd behaviors:
 - All readers do not always trigger a page load when switching from one page to another. I sometimes had mangaeden simply switching URL/picture without loading the page again
   => This might mess-up the auto-tracking if not handled !

And finally, the next steps:
 - Simplify UI where possible
 - Improve UX in general
 - Allows the browse mode to redirect to the Comic/Source's page to go on reading

### CODE REVIEW

File: web-extensions/chrome/scripts/load-bookmarks.js [hunk 1]
```diff
@@ -1,16 +1,66 @@
...
+BmcMangaList.prototype.onAliasClick = function(ev) {
+    const comicLabel = ev.target;
+    console.log(`BmcSideBar: BmcMangaList: onAlias: Label=${ev.target.innerText} id=${ev.target.id} reader=${readerName} name=${comicName}`);
+    const handleAliasError = err => {
+        bmcEngine.removeEventListener(bmcEngine.events.alias.error, handleAliasError);
+        bmcEngine.removeEventListener(bmcEngine.events.alias.complete, handleAliasSuccess);
+        this.setMode(this.MODE_BROWSE);
+        showHideSidePanel();
+        notifyResult('Alias Comic', new Error('Could not alias current page to comicId ' + comicId + ': ' + err.message));
+    }
```
Comment (GuillaumeGomez) [inline web-extensions/chrome/scripts/load-bookmarks.js]: You never put ';' at the end of the declaration of a closure?

File: web-extensions/chrome/scripts/load-bookmarks.js [hunk 2]
```diff
@@ -141,11 +191,144 @@ function addEvents(mangaList) {
...
+    elem.style.backgroundColor = color;
+
+    /*
+     * /!\ NOTE IMPORTANT /!\
+     * Accessing a DOM property seems to force a redraw, preventing browser
+     * optimization on style settings (by batching updates) which might
+     * actually hide the transition.
+     * /!\ NOTE IMPORTANT /!\
+     */
+    void elem.offsetHeight;
```
Comment (GuillaumeGomez) [inline web-extensions/chrome/scripts/load-bookmarks.js]: Sounds quite complicated for doing a transition... Going through CSS and just changing the value of a CSS property wouldn't be easier? I'm pretty sure it doesn't impact rendering performances either.

File: web-extensions/chrome/scripts/load-bookmarks.js [hunk 3]
```diff
@@ -141,11 +191,144 @@ function addEvents(mangaList) {
...
+
+    // Add the transition effect _before_ changing the color
+    elem.classList.add('notif-transform');
+
+    // And finally set the color to get back to (original color)
+    // => This triggers the actual transition effect
+    elem.style.backgroundColor = origColor;
+
+    // Remove all mentions of transitions after 2 secs
+    setTimeout(() => {
```
Comment (GuillaumeGomez) [inline web-extensions/chrome/scripts/load-bookmarks.js]: And that would also allow to remove this.

File: web-extensions/chrome/scripts/load-bookmarks.js (hunk 1 above)
Comment (Joacchim) [reply web-extensions/chrome/scripts/load-bookmarks.js]: Mistake; I will add the semicolons :)

File: web-extensions/chrome/scripts/load-bookmarks.js (hunk 2 above)
Comment (Joacchim) [reply web-extensions/chrome/scripts/load-bookmarks.js]: Well; we have an origin and target color.
As far as I understand, to transition from `A` to `B`,  one needs To have color `A` set; have the CSS transition class set; and only then apply the change of color to `B`.
First; I don't know how To do that in CSS Only. Maybe theres a back-and-forth transition fonction I should use to simplify..
Second, as I understand, the browser allows itself to batch the CSS property changes, which compacts all the color changes from this function into a Noop. The trick here prevents that mechanism by forcing the browser To redraw in between.

Any working simplification suggestion is thus welcome :)

Comment (GuillaumeGomez) [reply web-extensions/chrome/scripts/load-bookmarks.js]: Huuuuum. I'll take a look later on but that sounds like something doable with CSS only (maybe I'm wrong though?).

**Comment by GuillaumeGomez** [comment]
I tried your registration process and I don't really like it: it's way too complicated.

**Comment by GuillaumeGomez** [comment]
Here's the look I had in mind (which is in my commit):

screenshot from 2018-11-17 00-19-13

Having the "+" button directly integrated into the sidebar seems like a must have. And like I said, I really think we should think about the process of adding a manga because currently it's very hard to follow. :)

**Comment by Joacchim** [comment]
I was more thinking of having the "+" visible even without the sidebar open, replacing completely the current inforbar's register button.
Now, integrating it in the sidebar is an equivalent in my opinion, while the sidebar is open (and in "register" mode).

**Comment by Joacchim** [comment]
I was more thinking of having the "+" from the sidebar visible even without the sidebar open, replacing completely the current infobar's register button.

Now, integrating it in the sidebar the way you did is an equivalent in my opinion to keeping it on the side. This must only be visible when the sidebar is in "register" mode, though.

Additionally, I really want to keep a register button visible when the sidebar is closed to make the UI more intuitive for lambda users: The symbolic and operational path are clearer that way.

**Comment by Joacchim** [comment]
Btw, I need to understand clearly what's too complicated in that registration process, because I see the following improvements possible:
 1. pre-filling the searchbox with a name extracted from the page
 2. allowing to use "Enter" to validate the registration - which is desirable - (and not only a click)
 3. Automagically do stuff which would not allow for user customization during registration:
     a. define the manga name
     b. Detect duplicates, and create new sources for existing manga on behalf of the user

Personally, I'm against 3, because I think it makes stuff like customizing the name less intuitive for lambda users, and it requires quite a bit of work on our part for an incomplete and imperfect result.

File: web-extensions/chrome/scripts/load-bookmarks.js (hunk 2 above)
Comment (Joacchim) [reply web-extensions/chrome/scripts/load-bookmarks.js]: Let's address it as issue #18

File: web-extensions/chrome/scripts/load-bookmarks.js (hunk 3 above)
Comment (Joacchim) [reply web-extensions/chrome/scripts/load-bookmarks.js]: similarly, to be addressed as issue #18
